### PR TITLE
MC: Add protocol definitions for member management operations

### PR DESCRIPTION
**Pull request by emre-aydin**
Core PR: https://github.com/hazelcast/hazelcast/pull/15916

### CODE REVIEW

File: protocol-definitions/MC.yaml [hunk 1]
```diff
@@ -222,3 +222,96 @@ methods:
             8 - FREE_NATIVE_MEMORY_SIZE
             9 - FREE_NATIVE_MEMORY_PERCENTAGE
     response: {}
+  - id: 5
+    name: getMemberConfig
+    since: 2.0
+    doc: |
+      Gets the effective config of a member rendered as XML.
```
Comment (asimarslan) [inline protocol-definitions/MC.yaml]: What about YAML?

File: protocol-definitions/MC.yaml [hunk 2]
```diff
@@ -222,3 +222,96 @@ methods:
...
+    doc: |
+      Shuts down a member.
+    request:
+      retryable: false
+      acquiresResource: false
+      partitionIdentifier: -1
+      params: []
+    response: {}
+  - id: 9
+    name: promoteMember
```
Comment (asimarslan) [inline protocol-definitions/MC.yaml]: as we are promoting the lite member, the name should be whether `protomoteLiteMember` or `promoteToMember`

File: protocol-definitions/MC.yaml (hunk 1 above)
Comment (emre-aydin) [reply protocol-definitions/MC.yaml]: We only show the effective configuration. AFAIK it is only available as XML.

File: protocol-definitions/MC.yaml (hunk 2 above)
Comment (emre-aydin) [reply protocol-definitions/MC.yaml]: `promoteLiteMember` makes sense, will update accordingly.

File: protocol-definitions/MC.yaml [hunk 3]
```diff
@@ -222,3 +222,96 @@ methods:
...
+  - id: 5
+    name: getMemberConfig
+    since: 2.0
+    doc: |
+      Gets the effective config of a member rendered as XML.
+    request:
+      retryable: true
+      acquiresResource: false
+      partitionIdentifier: -1
+      params: []
```
Comment (asimarslan) [inline protocol-definitions/MC.yaml]: For an empty param list, you can remove the params definition altogether.
example: [Client.ping](https://github.com/hazelcast/hazelcast-client-protocol/blob/master/protocol-definitions/Client.yaml#L600)

File: protocol-definitions/MC.yaml [hunk 4]
```diff
@@ -222,3 +222,96 @@ methods:
...
+    response:
+      params:
+        - name: threadDump
+          type: String
+          nullable: false
+          since: 2.0
+          doc: |
+            Thread dump of the member's JVM.
+  - id: 8
+    name: shutdownMember
```
Comment (asimarslan) [inline protocol-definitions/MC.yaml]: How does this method decide which member to shutdown?
From a protocol perspective, it's not clear. A unisocket or smart client may have different behaviour.
A member uuid parameter or a better documentation can be added.

Comment (emre-aydin) [reply protocol-definitions/MC.yaml]: `Shuts down the member it's called on.` -> is this clear enough as documentation? Will add the same to the other which have the same behavior.

Comment (asimarslan) [reply protocol-definitions/MC.yaml]: Yes the definition is clear for me.
In order to shut down a member,  you'll need to open a connection to it. That drops any unisocket support. 
If that's the case, we may put a note at the top as this service only supports smart client.

File: protocol-definitions/MC.yaml [hunk 5]
```diff
@@ -222,3 +222,91 @@ methods:
...
+  - id: 9
+    name: promoteLiteMember
+    since: 2.0
+    doc: |
+      Promotes a lite member to a data member.
+    request:
+      retryable: false
+      acquiresResource: false
+      partitionIdentifier: -1
+    response: {}
```
Comment (ihsandemir) [inline protocol-definitions/MC.yaml]: is the result of the promotion notified by exception only if it failed?

File: protocol-definitions/MC.yaml (hunk 4 above)
Comment (emre-aydin) [reply protocol-definitions/MC.yaml]: We only use smart client. Will add the note to the service about unisocket client.

File: protocol-definitions/MC.yaml (hunk 5 above)
Comment (emre-aydin) [reply protocol-definitions/MC.yaml]: yes

File: protocol-definitions/MC.yaml (hunk 4 above)
Comment (emre-aydin) [reply protocol-definitions/MC.yaml]: Thinking again, I don't see a reason why we should drop support for unisocket clients. We might need them for whatever reason in the future. I will add the `member` parameter to the operations that need it (also to the ones I added in my previous PR).

Comment (emre-aydin) [reply protocol-definitions/MC.yaml]: After a discussion with Asim, we have decided to only document the requirement of smart client and keep the protocol definitions as they are.

Comment (emre-aydin) [reply protocol-definitions/MC.yaml]: I've updated the documentation.

**Pull request merged**